File: creational/singleton/java/realLife/SingletonJavaRealWorld/src/ClientCode.java
public class ClientCode {
  public static void main(String[] args) {
    final EventService eventService = EventService.getInstance();
    
    final Unsubscribe eventSubscription = eventService.subscribe(new Listener() {
      @Override
      public void invoke(String value) {
        System.out.println(value);
      }
    });

    final Unsubscribe eventSubscription2 = eventService.subscribe(new Listener() {
      @Override
      public void invoke(String value) {
        System.out.println(value + "eventSubscription2");
      }
    });
    
    eventService.emit("First event");
    eventService.emit("Second event");
    
    eventSubscription.unsubscribe();

    eventService.emit("Third event");
    eventService.emit("Fourth event");
    
    eventSubscription2.unsubscribe();
  }
}

File: design-patterns/creational/singleton/swift/realLife/Sources/singletonRealLife/clientCode.swift
@main
public struct clientCode {
    public static func main() {
      let eventService = EventService.shared;

      let eventSubscription = eventService.subscribe { value in
        print(value);
      }

      let eventSubscription2 = eventService.subscribe { value in
        print("\(value) eventSubscription2");
      }

      eventService.emit(value: "First event");
      eventService.emit(value: "Second event");

      eventSubscription.unsubscribe();

      eventService.emit(value: "Third event");

      eventService.emit(value: "Fourth event");

      eventSubscription2.unsubscribe();

    }
}
